data.rs

```diff
@@ -199,4 +199,48 @@ impl PSP22Data {
             amount,
         }])
     }
+
+    #[cfg(feature = "mintable")]
+    pub fn mint(&mut self, account: AccountId, value: u128) -> Result<Vec<PSP22Event>, PSP22Error> {
```

Comment:
I have two doubts:
1) Why does this return events instead of emiting them? I'm not saying it's bad, just trying to understand.
2) For `mint` and `burn` there needs to be some access control to these methods. Because currently anyone can mint and burn tokens on any account :P

Reply:
The idea behind this crate is to have the entire PSP22 token logic enclosed in a regular Rust struct, that can be later used in your contract in a way shown in `lib.rs`, kinda like a library providing basic building blocks of a token. It's up to contract's author to use these building blocks correctly.

1. Events are not emitted here, because that can only happen from within a contract, and this code is outside.
2. I feel like access control is not a part of token logic, but rather business logic of whatever contract uses this implementation. Whoever uses `PSP22Data` with mintable extension in their contact would need to decide under what circumstances the `mint()` method can be called.